ingestion/src/metadata/ingestion/ometa/mixins/patch_mixin.py

```diff
@@ -0,0 +1,197 @@
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Mixin class containing PATCH specific methods
+
+To be used by OpenMetadata class
+"""
+import json
+import traceback
+from typing import Generic, Optional, Type, TypeVar, Union
+
+from pydantic import BaseModel
+
+from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.type import basic
+from metadata.ingestion.ometa.client import REST
+from metadata.ingestion.ometa.utils import model_str, ometa_logger
+
+logger = ometa_logger()
+
+T = TypeVar("T", bound=BaseModel)  # pylint: disable=invalid-name
+
+OPERATION = "op"
+PATH = "path"
+VALUE = "value"
+
+# Operations
+ADD = "add"
+REPLACE = "replace"
+
+# OM specific description handling
+ENTITY_DESCRIPTION = "/description"
+COL_DESCRIPTION = "/columns/{index}/description"
+
+
+class OMetaPatchMixin(Generic[T]):
+    """
+    OpenMetadata API methods related to Tables.
+
+    To be inherited by OpenMetadata
+    """
+
+    client: REST
+
+    def _validate_instance_description(
+        self, entity: Type[T], entity_id: Union[str, basic.Uuid], force: bool = False
+    ) -> Optional[T]:
+        """
+        Validates if we can update a description or not. Will return
+        the instance if it can be updated. None otherwise.
+
+        Args
+            entity (T): Entity Type
+            entity_id: ID
+            description: new description to add
+            force: if True, we will patch any existing description. Otherwise, we will maintain
+                the existing data.
+        Returns
+            instance to update
+        """
+
+        instance = self.get_by_id(entity=entity, entity_id=entity_id)
+
+        if not instance:
+            logger.warn(
+                f"Cannot find an instance of {entity.__class__.__name__} with the given ID."
+            )
+            return
+
+        if instance.description and not force:
+            logger.warn(
+                f"The Entity already has a description. To overwrite it, set `force` to True."
+            )
+            return
+
+        return instance
+
+    def patch_description(
+        self,
+        entity: Type[T],
+        entity_id: Union[str, basic.Uuid],
+        description: str,
+        force: bool = False,
+    ) -> Optional[T]:
+        """
+        Given an Entity type and ID, JSON PATCH the description.
+
+        Args
+            entity (T): Entity Type
+            entity_id: ID
+            description: new description to add
+            force: if True, we will patch any existing description. Otherwise, we will maintain
+                the existing data.
+        Returns
+            Updated Entity
+        """
+        instance = self._validate_instance_description(
+            entity=entity, entity_id=entity_id, force=force
+        )
+        if not instance:
+            return
```

Comment:
I think we should return `None` explicitly just as a visual cue that we have a different path and that we are not breaking out of a loop